src/writers/Oaipmh.php

```diff
@@ -133,6 +133,8 @@ public function writeMetadataFile($metadata, $path, $overwrite = true)
     public function normalizeFilename($string)
     {
         $string = urldecode($string);
+        $pattern = '#'.DIRECTORY_SEPARATOR.'#';
```

Comment:
A bit more end-of-the-day hacking around has produced:

```php
    /**
     * Convert %3A (:) and other characters in filenames into underscores (_).
     */
    public function normalizeFilename($string)
    {
        $string = urldecode($string);
        $bad_chars = [
            ':' => '_',
            DIRECTORY_SEPARATOR => '_',
        ];
        $string = str_replace(array_keys($bad_chars), $bad_chars, $string);
        return $string;
    }
```
This replaces `:` in my local tests. Could you test to see if it catches the directory separator? The benefit of `str_replace()` is that you don't need to worry about nasty cross-platform issues (I hope).